### PR TITLE
chore: Add v1 APIs for v0.33.x

### PR DESCRIPTION
**Pull request by engedaam**
Fixes #N/A 

**Description**
- Add v1 APIs
- This is a copy and from main to release a patch version of 0.33

**How was this change tested?**
- `make presubmit`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/apis/v1/nodeclaim_status.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+	"github.com/awslabs/operatorpkg/status"
+	v1 "k8s.io/api/core/v1"
+)
+
+const (
+	ConditionTypeLaunched             = "Launched"
+	ConditionTypeRegistered           = "Registered"
+	ConditionTypeInitialized          = "Initialized"
+	ConditionTypeEmpty                = "Empty"
+	ConditionTypeDrifted              = "Drifted"
```
Comment (njtran) [inline pkg/apis/v1/nodeclaim_status.go:27]: I think you need to update this with the ConsolidateAfter changes, but you could consider doing that as a followup

File: pkg/apis/apis.go [hunk 2]
```diff
@@ -18,16 +18,18 @@ import (
 	_ "embed"
 
 	"github.com/samber/lo"
-	v1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 
+	v1 "sigs.k8s.io/karpenter/pkg/apis/v1"
```
Comment (njtran) [inline pkg/apis/apis.go:24]: Do you have to alias this? 
```suggestion
	"sigs.k8s.io/karpenter/pkg/apis/v1"
```

File: pkg/apis/v1/nodeclaim_status.go (hunk 1 above)
Comment (engedaam) [reply pkg/apis/v1/nodeclaim_status.go:28]: I added those changes

**Review by njtran — COMMENTED** [review]
/lgtm
/approve

**Review by njtran — APPROVED** [review]

**Pull request merged**